packages/xstate-wallet/src/chain.ts

```diff
@@ -248,8 +249,19 @@ export class ChainWatcher implements Chain {
 
   public async ethereumEnable(): Promise<string> {
     if (window.ethereum) {
-      const [selectedAddress] = await window.ethereum.enable();
-      return selectedAddress;
+      try {
+        this.selectedAddress = (await window.ethereum.enable())[0];
+        if (typeof this.selectedAddress === 'string') {
+          return this.selectedAddress;
+        } else {
+          console.error('Ethereum enabled but no selected address is defined');
+          return Promise.reject('Ethereum enabled but no selected address is defined');
```

Comment:
Out of interest, how / when would this happen?

Reply:
It shouldn't ever happen, as long as `window.ethereum.enable()` returns a promise that resolves to an array of strings. But we have no typings in place to be sure of that. 

I suppose we could just assert the type here, but dealing with the case where it resolves to something unexpected means we are building to the interface that we have, rather than the current implementation of metamask.